npc_gzip/distance.py

```diff
@@ -0,0 +1,211 @@
+from typing import Union
+
+import numpy as np
+
+from npc_gzip.exceptions import CompressedValuesEqualZero, InvalidShapeException
+
+
+class Distance:
+    def __init__(
+        self,
+        compressed_values_a: Union[list, np.ndarray],
```

Comment:
Is it necessary to use `np.ndarray`? The inputs to the distance metric like NCD are always 3 integers - the compressed lengths of training sample, test sample and concat samples.

Reply:
It's not completely necessary, lines 15-22 will do the conversion to numpy arrays anyways. 

The reason the init object needs to accept a list-like object is because it uses the methods like `_ncd` to calculate the ncd on the float values, while `@property.ncd` uses that method as a np.vectorized function.

Reply:
> Could `Sequence` (from `collections.abc`) be used instead of `Union[list, np.ndarray]`?

Good idea, updated!